osm2gtfs/core/helper.py

```diff
@@ -0,0 +1,112 @@
+# coding=utf-8
+
+import sys
+from math import cos, sin, atan2, sqrt, radians, degrees
+
+
+class Helper(object):
+    """The Helper class contains useful static functions
+
+    """
+
+    @staticmethod
+    def print_shape_for_leaflet(shape):
+        print "var shape = [",
+        i = 0
+        for node in shape:
+            print "new L.LatLng(" + str(node["lat"]) + ", " + str(node["lon"]) + ")",
+            if i != len(shape) - 1:
+                print ",",
+            i += 1
+        print "];"
+        i = 0
+        for node in shape:
+            print "L.marker([" + str(node["lat"]) + ", " + str(node["lon"]) + "]).addTo(map)"
+            print "    .bindPopup(\"" + str(i) + "\").openPopup();"
+            i += 1
+
+    @staticmethod
+    def get_center_of_nodes(nodes):
+        """Helper function to get center coordinates of a group of nodes
+
+        """
+        x = 0
+        y = 0
+        z = 0
+
+        if len(nodes) < 1:
+            sys.stderr.write("Cannot find the center of zero nodes\n")
+        for node in nodes:
+            lat = radians(float(node.lat))
+            lon = radians(float(node.lon))
+
+            x += cos(lat) * cos(lon)
+            y += cos(lat) * sin(lon)
+            z += sin(lat)
+
+        x = float(x / len(nodes))
+        y = float(y / len(nodes))
+        z = float(z / len(nodes))
+
+        center_lat = degrees(atan2(z, sqrt(x * x + y * y)))
+        center_lon = degrees(atan2(y, x))
+
+        return center_lat, center_lon
+
+    @staticmethod
+    def get_hex_code_for_color(color):
+        color = color.lower()
+        if color == u'black':
+            hex_color = '000000'
+        elif color == u'blue':
+            hex_color = '0000FF'
+        elif color == u'gray':
```

Comment:
We could support `grey` here too, couln't we?

Reply:
Changed to **webcolors**. (i hope) they do everything with a magical `name_to_hex` function.

Reply:
Actually I had some thoughts about the whole part of colors. And it turns out to be specific. Now, only Estelí is using it and especially the complementary color function I am not very convinced that it is mature to get into the "Core". So, I moved it to a newly created [routes_creator_esteli.py](https://github.com/mapanica/osm2gtfs/blob/master/osm2gtfs/creators/esteli/routes_creator_esteli.py).

Reply:
I'm not sure if this has to be specific. At least for Managua, the same approach can be used. It is very common to tag a route with the [`colour`](https://wiki.openstreetmap.org/wiki/Key:colour) tag, see the [OSM-Wiki](https://wiki.openstreetmap.org/wiki/Relation:route#Tags).

What exactly are your concerns about the complementary color computation?

Reply:
Yes, the `colour` tag from OpenSrreetMap is supported and for example Managua uses it (using hex values as theyal are also specified on OSM for the `colour` tag). I just suggested to move two things to the Estelí creator:

* conversion from name to hex: about the introduced function to map names and hex I was not convinced, so I thought webcolors was a good idea. But it is another dependency and I'm not sure really most creators may use it. Currently it is only Estelí. @grote suggested to do this in a separate issue.
* generation of a complementary color: first it wasn't working... The result was always #000000. The I checked it briefly, and realized that I'm doubting the "complementary" color is what we need. I think we need more the color with the highest contrast, which seems to be something different. I think it is not that easy than the function proposes and I would prefer to handle it also in a new issue.

Reply:
I opened up an issue to discuss this: #104